modules/currency-l1/src/main/scala/org/tessellation/currency/l1/domain/snapshot/programs/CurrencySnapshotProcessor.scala

```diff
@@ -224,7 +223,22 @@ object CurrencySnapshotProcessor {
               .map(_.flatMap(_.toOption))
               .map(NonEmptyList.fromList)
               .map(_.map(_.sortBy(_.value.ordinal)))
-              .flatMap(_.map(_.traverse(_.toHashedWithSignatureCheck)).sequence)
+              .flatMap(_.map(_.traverse { snapshot =>
+                def usingKryo = {
+                  implicit val hasher = Hasher.forKryo[F]
+                  snapshot.toHashedWithSignatureCheck
+                }
+
+                def usingJson = {
+                  implicit val hasher = Hasher.forJson[F]
+                  snapshot.toHashedWithSignatureCheck
+                }
+
+                usingJson.flatMap {
+                  case Left(_)  => usingKryo
+                  case Right(s) => s.asRight[Signed.InvalidSignatureForHash[CurrencyIncrementalSnapshot]].pure[F]
+                }
+              }).sequence)
```

Comment:
Replace the `_.map(_.traverse { snapshot => . . . }).sequence` with `_.traverse(_.traverse { snapshot => . . . })` ?